File: backend/app/schemas/article.py
from pydantic import BaseModel, HttpUrl, Field
from datetime import datetime
from typing import Optional
from typing import Optional, List, Tuple

class ArticleCreate(BaseModel):
    wikipedia_title: str
    wikipedia_url: HttpUrl
    processed_summary: str
    word_count: int
    frequent_words: List[Tuple[str, int]] = Field(default_factory=list)
    personal_notes: Optional[str] = None
    sentiment_polarity: Optional[float] = None
    sentiment_subjectivity: Optional[float] = None
    sentiment_label: Optional[str] = None
    user_id: int
    
class ArticleUpdate(BaseModel):
    personal_notes: Optional[str] = None

class ArticleInDB(ArticleCreate):
    id: int
    created_at: datetime | None = None
    saved_at: datetime | None = None

    class Config:
        from_attributes = True

File: frontend/src/types/article.ts
export interface ArticleCreate {
  wikipedia_title: string;
  wikipedia_url: string; 
  processed_summary: string;
  word_count: number;
  frequent_words: [string, number][];
  
  sentiment_polarity?: number | null;
  sentiment_subjectivity?: number | null;
  sentiment_label?: string | null;
  personal_notes?: string | null;

  user_id: number; 
}

export interface ArticleUpdate {
  personal_notes?: string | null;
}

export interface ArticleInDB extends ArticleCreate {
  id: number;
  created_at: string | null;
  saved_at: string | null;
}